packages/liveblocks-core/src/crdts/liveblocks-helpers.ts

```diff
@@ -208,18 +208,18 @@ export function getTreesDiffOperations(
           });
           break;
         case CrdtType.OBJECT:
-          ops.push(
-            crdt.parentId
-              ? {
-                  type: OpCode.CREATE_OBJECT,
-                  id,
-                  parentId: crdt.parentId,
-                  parentKey: crdt.parentKey,
-                  data: crdt.data,
-                }
-              : // Root object
-                { type: OpCode.CREATE_OBJECT, id, data: crdt.data }
-          );
+          if (crdt.parentId === undefined || crdt.parentKey === undefined) {
+            throw new Error(
+              "Internal error. Cannot serialize storage root into an operation"
+            );
+          }
```

Comment:
Any reason to add this runtime check here? TypeScript should have us covered, I think. Won't our linter nag about these checks being useless?

Reply:
`SerializedRootObject` does not have a `parentId` or a `parentKey`, but it can never be sent from the client. The root is initialized from the server. That's why we never have a `CreateRootObjectOp`.